node/actors/network/src/http/mod.rs

```diff
@@ -0,0 +1,280 @@
+    where
+        K: std::hash::Hash + Eq + Clone + std::fmt::Debug + zksync_consensus_crypto::TextFmt,
+    {
+        let mut table = Table::new()
+            .with_custom_header_row(
+                TableRow::new()
+                    .with_cell(TableCell::new(TableCellType::Header).with_raw("Public key"))
+                    .with_cell(TableCell::new(TableCellType::Header).with_raw("Address"))
+                    .with_cell(
+                        TableCell::new(TableCellType::Header)
+                            .with_attributes([("colspan", "2")])
+                            .with_raw("Incoming"),
+                    )
+                    .with_cell(
+                        TableCell::new(TableCellType::Header)
+                            .with_attributes([("colspan", "2")])
+                            .with_raw("Outgoing"),
+                    )
+                    .with_cell(TableCell::new(TableCellType::Header).with_raw("Age")),
+            )
+            .with_header_row(vec!["", "", "size", "bandwidth", "size", "bandwidth", ""]);
+        for (key, values) in connections {
+            let age = SystemTime::now()
+                .duration_since(values.established)
+                .ok()
+                .unwrap_or_else(|| Duration::new(1, 0))
+                .max(Duration::new(1, 0)); // Ensure Duration is not 0 to prevent division by zero
+            let received = values.received.load(std::sync::atomic::Ordering::Relaxed);
+            let sent = values.sent.load(std::sync::atomic::Ordering::Relaxed);
+            table.add_body_row(vec![
+                self.shorten(key),
+                values.address.map_or("-".to_string(), |a| a.to_string()),
+                bytesize::to_string(received, false),
+                bytesize::to_string(received / age.as_secs(), false) + "/s",
+                bytesize::to_string(sent, false),
+                bytesize::to_string(sent / age.as_secs(), false) + "/s",
+                format!("{}s", age.as_secs()),
+            ])
+        }
+        table.to_html_string()
+    }
+
+    fn shorten<K>(&self, key: K) -> String
+    where
+        K: std::fmt::Debug + zksync_consensus_crypto::TextFmt,
+    {
+        let key = key.encode();
+        key.strip_prefix("validator:public:bls12_381:")
+            .or(key.strip_prefix("node:public:ed25519:"))
+            .map_or("-".to_string(), |key| {
+                let len = key.len();
+                format!("{}...{}", &key[..10], &key[len - 11..len])
+            })
+    }
+
+    fn tls_acceptor(&self) -> TlsAcceptor {
+        let cert_der = load_certs(&self.cert_path).expect("Invalid certificate");
+        let key_der = load_private_key(&self.key_path).expect("Invalid private key");
+        Arc::new(
+            ServerConfig::builder()
+                .with_no_client_auth()
+                .with_single_cert(cert_der, key_der)
+                .unwrap(),
+        )
+        .into()
+    }
```

Comment:
these should have been loaded to the config directly and if possible verified upfront.